src/components/DroughtThresholds.vue

```diff
@@ -77,239 +77,179 @@
-              onLeave - scrolling down, end meets scroller-end
-              onEnterBack - scrolling up, end meets scroller-end
-              onLeaveBack - scrolling up, start meets scroller-start
-              */
-            },
-          })
-        }
-        if (i == (containers.length-1)) {
-          tl.to(`.${scrollClass}`, {
-            scrollTrigger: {
-              markers: marker_on,
-              trigger: `.${scrollClass}`,
-              start: "top 41%",
-              end: "top 41%",
-              onEnter: () => {
-                document.querySelector("#next").classList.remove("visible");
-              },
-              onLeaveBack: () => {
-                document.querySelector("#next").classList.add("visible");
-              }
-            },
-          })
-        }
-      })
-
-    });
-
-    function scrollFxn(e) {
-        const scrollButton = e.target; // define target
-        const scrollID = scrollButton.id; // extract id as "button-x"
-        const scrollFrame = scrollID.split('-')[1]; // extract frame number "x"
-        console.log(scrollID)
-      // scroll to position of specified frame
-        gsap.to(window, {duration: 0, scrollTo:"#scroll-to-"+scrollFrame});
-      }
-
-    function prevFxn(e) {
-        const currentFrame = document.querySelector('#svg .visible'); // get svg element that is visible
-        const currentFrameName = currentFrame.id; // full id name in format "step-x"
-        const currentFrameLetter = currentFrameName.split('-')[1]
-
-        const prevFrameLetter = String.fromCharCode(currentFrameLetter.charCodeAt(0) - 1); // prev letter
-
-        //scroll to previous
-        gsap.to(window, {duration: 0, scrollTo:"#scroll-to-" + prevFrameLetter})
+        })
       }
-
-      function nextFxn(e) {
-        const currentFrame = document.querySelector('#svg .visible'); // get svg element that is visible
-        const currentFrameName = currentFrame.id; // full id name in format "step-x"
-        const currentFrameLetter = currentFrameName.split('-')[1]
-
-        const nextFrameLetter = String.fromCharCode(currentFrameLetter.charCodeAt(0) + 1); // next letter
-        //scroll to next
-        gsap.to(window, {duration: 0, scrollTo:"#scroll-to-"+nextFrameLetter})
+      if (i == (containers.length-1)) {
+        tl.to(`.${scrollClass}`, {
+          scrollTrigger: {
+            markers: marker_on,
+            trigger: `.${scrollClass}`,
+            start: "top 41%",
+            end: "top 41%",
+            onEnter: () => {
+              document.querySelector("#next").classList.remove("visible");
+            },
+            onLeaveBack: () => {
+              document.querySelector("#next").classList.add("visible");
+            }
+          },
+        })
       }
+    })
 
-    return {
-      windowSizeStore,
-      marker_on,
-      frames,
-      w,h,margin,
-      scrollFxn,
-      prevFxn,
-      nextFxn,
-      DroughtCharts
```

Comment:
I did this because I thought it was necessary to access them in the template - especially frames, scrollFxn, prevFxn, and nextFxn

Reply:
It definitely was necessary w/ the original syntax, but not with the new composition API syntax

Reply:
To clarify - I think returning the defined functions was necessary with your initial approach (using `setup()` + `onMounted()` in place of `mounted()`). Returning them isn't necessary if we use the old syntax (`mounted()` + `methods()`), and also isn't necessary if we use the composition API syntax.